### PR TITLE
[Keyboard] New Keyboard - BDE Rev2

### PR DESCRIPTION
**Pull request by kylemccreery**
## Description

Initial PR for adding the BDE Rev2 keyboard kit by MechWild. This is the default and VIA keymaps. Additionally, I reorganized the original BDE folder for easier building and maintaining legacy version support (Rev1)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* None

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mechwild/bde/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  [_FN3] = LAYOUT(                                          
+    KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,    KC_F6,    KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, RGB_TOG,       
+    _______, _______, _______, _______, _______,  _______,  _______, _______, _______, _______, RGB_HUI, RGB_SAI, RGB_VAI, RGB_MOD,  
+    _______, _______, _______, _______, _______,  _______,  _______, _______, _______, _______, RGB_HUD, RGB_SAD, RGB_VAD, RGB_RMOD
+  )
+
+};
+
+
+#ifdef OLED_ENABLE
```
Comment (drashna) [inline keyboards/mechwild/bde/keymaps/via/keymap.c:62]: As should these.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mechwild/bde/keymaps/via/keymap.c (hunk 1 above)
Comment (kylemccreery) [reply keyboards/mechwild/bde/keymaps/via/keymap.c:62]: @drashna I don't see the first part of the message that assumedly this is a follow up to. Did a comment disappear?

Comment (drashna) [reply keyboards/mechwild/bde/keymaps/via/keymap.c:62]: Sorry.  Specifically, it should be that all of the oled functions should be moved into the keyboard c file, and switched to the kb function.  

That way, a default behavior can be set, but can be easily overridden, if desired.

**Comment by kylemccreery** [comment]
@drashna Thanks for the clarification. Fixed.

File: keyboards/mechwild/bde/rules.mk [hunk 2]
```diff
@@ -12,8 +12,11 @@ MOUSEKEY_ENABLE = yes       # Mouse keys
 EXTRAKEY_ENABLE = yes       # Audio control and System control
 CONSOLE_ENABLE = no         # Console for debug
 COMMAND_ENABLE = no         # Commands for debug and configuration
-NKRO_ENABLE = yes           # Enable N-Key Rollover
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/mechwild/bde/rules.mk:15]:
```suggestion
NKRO_ENABLE = yes            # Enable N-key Rollover
```

**Review by drashna — APPROVED** [review]

**Comment by kylemccreery** [comment]
@drashna Should a review be requested from qmk/collaborators now?

File: keyboards/mechwild/bde/rules.mk [hunk 3]
```diff
@@ -12,8 +12,11 @@ MOUSEKEY_ENABLE = yes       # Mouse keys
 EXTRAKEY_ENABLE = yes       # Audio control and System control
 CONSOLE_ENABLE = no         # Console for debug
 COMMAND_ENABLE = no         # Commands for debug and configuration
-NKRO_ENABLE = yes           # Enable N-Key Rollover
+NKRO_ENABLE = yes           # Enable N-key Rollover
```
Comment (fauxpark) [inline keyboards/mechwild/bde/rules.mk:15]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

File: keyboards/mechwild/bde/rules.mk [hunk 4]
```diff
@@ -12,8 +12,11 @@ MOUSEKEY_ENABLE = yes       # Mouse keys
...
 BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
 RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
 AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = no         # Enable encoder
+OLED_ENABLE = no            # Enable OLED Screen
```
Comment (fauxpark) [inline keyboards/mechwild/bde/rules.mk:20]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by kylemccreery** [comment]
@fauxpark is there another change that needs to be made?

File: keyboards/mechwild/bde/rules.mk [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# MCU name
+ # MCU name
```
Comment (fauxpark) [inline keyboards/mechwild/bde/rules.mk:1]:
```suggestion
# MCU name
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by kylemccreery** [comment]
@fauxpark is this just waiting on merging at this point?

**Comment by drashna** [comment]
Thanks!

**Pull request merged**